src/jit/codegencommon.cpp

```diff
@@ -10317,10 +10333,23 @@ void CodeGen::genFuncletProlog(BasicBlock* block)
 
     compiler->unwindBegProlog();
 
+    inst_RV(INS_push, REG_FPBASE, TYP_REF);
+    compiler->unwindPush(REG_FPBASE);
+
     // TODO Save callee-saved registers
 
     // This is the end of the OS-reported prolog for purposes of unwinding
     compiler->unwindEndProlog();
+
+    // Restore frame pointer
+    // TODO We will need changes here when we introduce PSPSym
+    inst_RV_RV(INS_mov, REG_FPBASE, REG_ARG_0);
```

Comment:
@wateret yes, the patch is exactly what I've meant. It is possible that it will fix more issues in the coreclr tests, since the pattern in the test is quite common.
As for pushing the Context_Eip on the stack - that enables the stack unwinder to unwind from ThrowExceptionHelper to CallEHFunclet. It creates a stack frame looking as if the ThrowExceptionHelper was called by the CallEHFunclet, which allows the exception to be propagated from ThrowExceptionHelper to CallEHFunclet and further.